src/python/pants/option/options.py

```diff
@@ -150,19 +150,19 @@ def create(
             scope: registrar.known_scoped_args for scope, registrar in registrar_by_scope.items()
         }
 
-        config_to_pass = None if native_options_config_discovery else config.sources()
```

Comment:
`native_options_config_discovery` is no more - as mentioned above, this is now done by passing `None` for config sources. And since we no longer have any python-side config file discovery logic, https://github.com/pantsbuild/pants/issues/21091 is no longer an issue.